### PR TITLE
WIP: move /cluster/images/conformance to /test/images/conformance

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

Move the source of the conformance image from /cluster/images/conformance to /test/images/conformance.

**Which issue(s) this PR fixes**:

xref https://github.com/kubernetes/kubernetes/issues/78995
related test-infra change: https://github.com/kubernetes/test-infra/pull/18799

**Special notes for your reviewer**:
NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: build/images/conformance/README.md [hunk 1]
```diff
@@ -37,5 +37,3 @@ If you don't want to push the images, run `make` or `make build` instead
 ```
 kubectl create -f conformance-e2e.yaml
 ```
-
-[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/cluster/images/conformance/README.md?pixel)]()
```
Comment (neolit123) [inline build/images/conformance/README.md:41]: this link is no longer valid. removed it.

**Comment by BenTheElder** [comment]
I'm not sure how we want to organize build/, currently we have e.g. build/pause and build/go-runner top-level

I think maybe build/images/pause would be better for example, but it isn't how things are currently organized

**Comment by BenTheElder** [comment]
This also maybe better belongs in test/... ?
The other images in build/ are critical to a functioning cluster or are base images.

**Comment by neolit123** [comment]
> I think maybe build/images/pause would be better for example, but it isn't how things are currently organized

seems good. all images under /build should be under /build/images.
this can be done in separate PRs.

> This also maybe better belongs in test/..., actually ?

this actually makes more sense. i can update.

**Comment by neolit123** [comment]
> This also maybe better belongs in test/..., actually ?

updated.

/assign @BenTheElder @dims

**Comment by neolit123** [comment]
> pull-kubernetes-conformance-image-test

this will likely fail until https://github.com/kubernetes/test-infra/pull/18799 merges.

**Comment by spiffxp** [comment]
I'm not interested in seeing this land for v1.19 but agree `cluster` is probably the wrong place for it

I'm less certain about `test/images` being the right place, if only because everything else in there, I think? is an image that is scheduled _by_ e2e tests.  The conformance image doesn't fit into that definition.

**Comment by neolit123** [comment]
> I'm less certain about test/images being the right place, if only because everything else in there, I think? is an image that is scheduled by e2e tests. The conformance image doesn't fit into that definition.

i think this is mostly correct.

one partial exception that i found is:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml#L23

but if /build/images or /test/images are not the right locations, i'm out of ideas.

**Comment by mikedanese** [comment]
Looks like some of this needs cleanup:

https://github.com/kubernetes/test-infra/blob/bec22c9286285b8f177b3436508c4dade18354ac/experiment/kind-conformance-image-e2e.sh#L121-L136

**Comment by BenTheElder** [comment]
> I'm less certain about test/images being the right place, if only because everything else in there, I think? is an image that is scheduled by e2e tests. The conformance image doesn't fit into that definition.

Fair. test/e2e/image then? Little about it is conformance specific, it's a docker-wrapped e2e.test.

I don't think it belongs to build.

It's also a little weird because the only major consumer of this image has remained a downstream vendor owned tool, outside of the CI rig we have, I'm not aware of evidence that this image is used in any other way.

Also agree that this can wait for 1.20.

**Comment by neolit123** [comment]
> Looks like some of this needs cleanup:
> https://github.com/kubernetes/test-infra/blob/bec22c9286285b8f177b3436508c4dade18354ac/experiment/kind-conformance-image-e2e.sh#L121-L136

yes, i've sent this yesterday:
https://github.com/kubernetes/test-infra/pull/18799

**Comment by neolit123** [comment]
> test/e2e/image then?

or /test/conformance/image

/assign @spiffxp

**Comment by desponda** [comment]
👋  Bug triage here, this MR has not had activity for a while, I am removing the milestone for this PR for now, please update if this needs to go into a future release

**Comment by spiffxp** [comment]
Thanks for removing from v1.21

/remove-lifecycle stale
/milestone v1.22

**Comment by neolit123** [comment]
doing cleanup on some of my old pending PRs. let's revisit this in the future.

**Pull request closed without merging**